apps/backend/app/api/routes/visualizations.py

```
@@ -0,0 +1,257 @@
+    dataset_id: str,
+    x_column: str,
+    y_column: str,
+    filters: Optional[str] = Query(None),
+    current_user_id: str = Depends(get_current_user_id),
+):
+    """Get scatter plot data for two columns"""
+    try:
+        # Get dataset
+        dataset = await UserData.get(dataset_id)
+        if not dataset or dataset.user_id != current_user_id:
+            raise HTTPException(status_code=404, detail="Dataset not found")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)